src/Safebox.sol

```diff
@@ -0,0 +1,238 @@
+// SPDX-FileCopyrightText: © 2022 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity ^0.8.16;
+
+/**
+ * @author amusingaxl
+ * @title A safebox for digital assets.
+ * @notice
+ * - The `owner` is in full control of how much and when it can send assets to `recipient`.
+ *   - If MakerDAO governance ever executes an Emergency Shutdown, anyone can send assets to `recipient`.       This prevents assets being stuck in this contract when the governance smart contract is no longer operational.
+ * - The `custodian` cooperation is required whenever the `owner` wants to update the `recipient`.
+ */
+contract Safebox {
+    /// @notice MCD Vat module.
+    VatLike public immutable vat;
+
+    /// @notice Addresses with owner access on this contract. `wards[usr]`
+    mapping(address => uint256) public wards;
+    /// @notice Addresses with custodian access on this contract. `can[usr]`
+    mapping(address => uint256) public can;
```

Comment:
If custodian is effectively immutable and can only be set in the constructor why is this mapping needed?

Reply:
Yeah this is a good observation. Let's just make an immutable variable and remove the mapping.

Reply:
Coinbase requested the ability to be able to rotate their keys, so this will be needed. I'm adding the functionality right now.